Neuron/Sections/Wallet/Assets/AddAssetController.swift

```diff
@@ -8,159 +8,198 @@
-        case 3:
-            cell.isEdit = false
-            cell.rightTextField.text = tokenModel.symbol
-        case 4:
-            cell.isEdit = false
-            cell.rightTextField.text = tokenModel.decimals == 0 ? "0" : String(tokenModel.decimals)
-        default:
-            break
+            let cell = tableView.dequeueReusableCell(withIdentifier: "selectChainTableViewCell") as! SelectChainTableViewCell
+            cell.detailTextLabel?.text = chain.chainName
+            return cell
+        } else {
+            let cell = tableView.dequeueReusableCell(withIdentifier: "contractAddressTableViewCell") as! ContractAddressTableViewCell
+            cell.delegate = self
+            cell.contractAddressTextField.text = inputText
+            if chain.chainId == SwitchChainViewController().appChainId {
+                cell.contractAddressTextField.placeholder = "Assets.AddAssets.NodeAddressPlaceHolder".localized()
+                cell.contractAddressLabel.text = "Assets.AddAssets.NodeAddress".localized()
+            } else {
+                cell.contractAddressLabel.text = "Assets.AddAssets.ContractAddress".localized()
+                cell.contractAddressTextField.placeholder = "Assets.AddAssets.ContractAddressPlaceHolder".localized()
+            }
+            return cell
         }
-        return cell
     }
 
-    func tableView(_ tableView: UITableView, willSelectRowAt indexPath: IndexPath) -> IndexPath? {
-        return nil
-    }
-
-    func didClickSelectCoinBtn() {
-        nView.frame = CGRect(origin: .zero, size: UIScreen.main.bounds.size)
-        nView.delegate = self
-        nView.dataArray = [["name": "以太坊eth", "id": "100"], ["name": "test-chain", "id": "101"]]
-        nView.selectDict = ["name": "以太坊eth", "id": "100"]
-        UIApplication.shared.keyWindow?.addSubview(nView)
-    }
+    @IBAction func clickSelectChainButton(_ sender: UIButton) {
 
-    func callBackDictionnary(dict: [String: String]) {
     }
 
-    func didClickQRCodeBtn() {
+    @IBAction func clickQRCodeButton(_ sender: UIButton) {
         let qrCodeViewController = QRCodeViewController()
         qrCodeViewController.delegate = self
         self.navigationController?.pushViewController(qrCodeViewController, animated: true)
     }
 
-    func didBackQRCodeMessage(codeResult: String) {
-        tokenModel.address = ""
-        let finalText = codeResult.replacingOccurrences(of: " ", with: "")
-        tokenModel.address = finalText
-        if finalText.count == 40 || finalText.count == 42 {
-            didGetERC20Token(token: finalText)
+    func ethereumERC20Token(contractAddress: String) {
+        let walletAddress = AppModel.current.currentWallet!.address
+        Toast.showHUD()
+        DispatchQueue.global().async {
+            let result = try? CustomERC20TokenService.searchTokenData(contractAddress: contractAddress, walletAddress: walletAddress)
+            DispatchQueue.main.async {
```

Comment:
It's better to use `[weak self]` here even though toast will block user interaction. (toast behavior might change in the future)